textdb/textdb-dataflow/src/test/java/edu/uci/ics/textdb/dataflow/join/JoinDistanceHelper.java

```diff
@@ -0,0 +1,139 @@
+        RelationManager relationManager = RelationManager.getRelationManager();
+        for (int i = 0; i < tuples.length; i++) {
+            relationManager.insertTuple(BOOK_TABLE_INNER, tuples[i]);
+        }
+    }
+    
+    public static void insertToInner(List<ITuple> tuples) throws StorageException {
+        RelationManager relationManager = RelationManager.getRelationManager();
+        for (ITuple tuple: tuples) {
+            relationManager.insertTuple(BOOK_TABLE_INNER, tuple);
+        }
+    } 
+    
+    public static void clearTestTables() throws TextDBException {
+        RelationManager relationManager = RelationManager.getRelationManager();
+        relationManager.deleteTuples(BOOK_TABLE_OUTER, new MatchAllDocsQuery());
+        relationManager.deleteTuples(BOOK_TABLE_INNER, new MatchAllDocsQuery());      
+    }
+    
+    public static void deleteTestTables() throws TextDBException {
+        RelationManager relationManager = RelationManager.getRelationManager();
+        relationManager.deleteTable(BOOK_TABLE_OUTER);
+        relationManager.deleteTable(BOOK_TABLE_INNER);
+    }
+    
+    public static KeywordMatcherSourceOperator getKeywordSource(String tableName, String query, 
+            KeywordMatchingType matchingType) throws TextDBException {
+        DataStore tableDataStore = RelationManager.getRelationManager().getTableDataStore(tableName);
+        KeywordPredicate keywordPredicate = new KeywordPredicate(query, 
+                Arrays.asList(JoinTestConstants.AUTHOR, JoinTestConstants.TITLE, JoinTestConstants.REVIEW),
+                LuceneAnalyzerConstants.getStandardAnalyzer(), matchingType);
+        KeywordMatcherSourceOperator keywordSource = new KeywordMatcherSourceOperator(keywordPredicate, tableDataStore);
+        return keywordSource;
+    }
+    
+    public static List<ITuple> getJoinDistanceResults(IOperator outerOp, IOperator innerOp,
+            String idAttrName, String joinAttrName, int threshold, int limit, int offset) throws TextDBException {
+        JoinDistancePredicate distancePredicate = new JoinDistancePredicate(
+                idAttrName, joinAttrName, threshold);
+        Join join = new Join(distancePredicate);
+        join.setInnerInputOperator(innerOp);
+        join.setOuterInputOperator(outerOp);
+        join.setLimit(limit);
+        join.setOffset(offset);
+        
+        ITuple tuple;
+        List<ITuple> results = new ArrayList<>();
+        
+        join.open();
+        while ((tuple = join.getNextTuple()) != null) {
+            results.add(tuple);
+        }
+        join.close();
+        
+        return results;
+    }
+    
+    /**
+     * Alter a field of a tuple. The schema will also be changed accordingly.
+     * 
+     * @param originalTuple
+     * @param fieldIndex
+     * @param newField
+     * @return
+     */
+    public static ITuple alterField(ITuple originalTuple, int fieldIndex, IField newField) {
```

Comment:
This helper function modifies a field in tuple (which is needed in some test cases).